compiler/optimizations/bulkCopyRecords.cpp

```diff
@@ -135,11 +44,25 @@ static bool isAssignment(FnSymbol* fn)
 }
 
 
-static bool isTrivialAssignment(FnSymbol* fn, FnSet& visited)
+static bool isTrivialAssignment(FnSymbol* fn)
 {
   if (! isAssignment(fn))
     return false;
-  if (isTrivial(fn, visited))
+  
+  // The base argument types must match.
+  ArgSymbol* lhs = fn->getFormal(1);
+  ArgSymbol* rhs = fn->getFormal(2);
+  if (lhs->type->getValType() != rhs->type->getValType())
+    return false;
+
+  // Don't consider assignment for class types to be trivial.
+  // That is because a class type represents a pointer, so
+  // we don't really track whether or not it is POD.
```

Comment:
and the test is:
test/classes/hilde/freeAssignment.chpl

At the moment we are not set on supporting the code that's in the test, so it is TBD.